src/runtime/impure/lfortran_intrinsics.c

```diff
@@ -606,7 +606,10 @@ LFORTRAN_API void _lfortran_strcat(char** s1, char** s2, char** dest)
 
 // idx1 and idx2 both start from 1
 LFORTRAN_API char* _lfortran_str_copy(char* s, int32_t idx1, int32_t idx2) {
-    char* dest_char = (char*)malloc(idx2-idx1+2);
+    if (idx2 == idx1) {
+        idx2 = idx1 + 1;
+    }
+    char* dest_char = (char*)malloc(idx2-idx1+1);
     for (int i=idx1; i <= idx2; i++)
```

Comment:
```suggestion
    for (int i=idx1; i < idx2; i++)
```
With this change, all the tests in https://github.com/lcompilers/lpython/issues/273#issue-1176977053 will work. Right now they don't -- and the build is a success because string comparison is not handled properly - 

https://github.com/lcompilers/lpython/blob/c251f6427cd2671353d8264a6b82c087e17981ae/src/libasr/codegen/asr_to_llvm.cpp#L2646

With the latest commit, I get -
```console
(lp) namannimmo@srf lpython % cat t.py               
def main():
  s: str
  s = "012345"
  print(s[2:4])
  print(s[1:6])
  print(s[0:6])
  s = "abcdefg"
  print(s[1])
  print(s[0])
  print(s[1:2])
  print(s[1:3])

main()

(lp) namannimmo@srf lpython % lpython t.py && ./a.out
234
12345
012345
bc
ab
bc
bcd
```

Reply:
Good catch! I think I missed that. Seems like string comparison doesn't work well.